### PR TITLE
Fix "nil pointer dereference"

### PR DESCRIPTION
**Pull request by difince**
fixing " nil pointer dereference" error on "./edgex-cli version" command
when incorrect IP is provided.
goimports, gofmt
Start using cobra "RunE" - Run method that return error
Start using cmd.Println instead of fmt.Pringln

Fixes: https://github.com/edgexfoundry-holding/edgex-cli/issues/137

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: pkg/cmd/version/version.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+//TODO move Version struct into go-mod-core-contracts
+type Version struct {
+	Version string `json:"version" yaml:"version,omitempty"`
+}
+
+// GetVersion returns the version of core-data microservice
+func GetVersion(port string) (version Version, err error) {
+	host := viper.GetString("dataservice.host")
+	url := "http://" + host + ":" + port + "/api/version"
```
Comment (difince) [inline pkg/cmd/version/version.go:18]: Just to note: `url := "http://" + host + ":" + port + "/api/version"`
Here, I am not useing "net.url" package. I just construct the URL string 
on the next line inside `http.Get(url)` `net.url.Parse` function is called - to validate that the URL is valid one. No need to do it twice.

File: cmd/version/version.go [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
 		Short: "Version command",
 		Long:  `Outputs the current version of edgex-cli.`,
-		Run: func(cmd *cobra.Command, args []string) {
-			// Printing the CLI's version
-			fmt.Println("edgex-cli version: ", version)
-			// printing the version of each service
-			data := client.GetVersion("48080")
-			fmt.Println("version: ", string(data))
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			version, err := version.GetVersion("48080")
```
Comment (mkbhanda) [inline cmd/version/version.go:16]: Earlier it printed out two versions, the edgex-cli version and the edgex-go version.
Now only edgex-go version. Need also the contents of VERSION file in the edgex-cli repo. Unclear to me how that got fetched earlier just looking at the code.

File: cmd/version/version.go [hunk 3]
```diff
@@ -1,26 +1,24 @@
...
 )
 
-var version string
-
 // NewCommand returns the version command
 func NewCommand() *cobra.Command {
 	var cmd = &cobra.Command{
 		Use:   "version",
 		Short: "Version command",
 		Long:  `Outputs the current version of edgex-cli.`,
```
Comment (mkbhanda) [inline cmd/version/version.go:14]: The documentation does not match.
s/edgex-cli/edgex-cli and edgex-go

File: pkg/cmd/version/version.go (hunk 1 above)
Comment (mkbhanda) [reply pkg/cmd/version/version.go:18]: perfect. a clean up task for the future.

File: cmd/version/version.go (hunk 2 above)
Comment (difince) [reply cmd/version/version.go:16]: Really .. let me check what I have wiped it out

File: cmd/version/version.go (hunk 3 above)
Comment (difince) [reply cmd/version/version.go]: I have changed the Long message to : 
`Outputs the current versions of EdgeX CLI and EdgeX Foundry`

And in addition, I have changed the banner to - "EdgeX-CLI"
and the version of the CLI is immediately printed under the banner

**Comment by mkbhanda** [comment]
Odd, but still get two error message print outs when I try with fictitious IP for edgeX host

**Pull request merged**